Sming/SmingCore/Network/TcpConnection.cpp

```diff
@@ -165,7 +165,8 @@ void TcpConnection::onError(err_t err)
 #ifdef ENABLE_SSL
 	if(ssl) {
 		sslConnected = false;
-		ssl_free(ssl); // ssl_free frees internally also the SSL context and the SSL extension data
+		ssl_free(ssl);
+		ssl_ctx_free(sslContext);
```

Comment:
Hm.. looking at the axtls code for `ssl_ctx_free` it seems that the function internally calls `ssl_free`. Excerpt:

```C
EXP_FUNC void STDCALL ssl_ctx_free(SSL_CTX *ssl_ctx)
{
    SSL *ssl;
    int i;

    if (ssl_ctx == NULL)
        return;

    ssl = ssl_ctx->head;

    /* clear out all the ssl entries */
    while (ssl)
    {
        SSL *next = ssl->next;
        ssl_free(ssl);
        ssl = next;
    }

// ...
```

Which means that probably just replacing `ssl_free(ssl);` with `ssl_ctx_free(sslContext);` should work.

Reply:
that's true, I'll try :)

Reply:
@ericogr May be one last thing - add the following comment behind the `ssl_ctx_free(sslContext);` // ssl_ctx_free frees the context. Internally it calls ssl_free which frees SSL extension data and SSL itself.